.github/workflows/nix.yml

```diff
@@ -0,0 +1,59 @@
+name: Nix
+
+on:
+  - pull_request
+  - push
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: cachix/install-nix-action@v17
+      - run: nix flake check --no-build
+
+  build-static:
+    name: Build static binary
+    needs: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: cachix/install-nix-action@v17
+      - uses: cachix/cachix-action@v10
+        with:
+          name: rushmorem # TODO change this to SurrealDB binary cache
```

Comment:
@tobiemh This PR is finally ready, except for this last bit of course. A binary cache is totally optional but very beneficial, especially since it takes a while to build this project.

Cachix is widely used in the Nix open source community as it comes with a free 10 GB package for open source projects.  Their website says 5 GB but my account is up to 9.62 GB so far and part of the email I received from them yesterday reads:-

> You've uploaded 8.60 GB of your current storage limit 10.00 GB.
>
> You have two options:
>
> a) Do nothing and least recently downloaded cache entries will be deleted to maintain your current storage limit.
>   See https://app.cachix.org/garbage-collection for details.
>
> b) Upgrade your plan at https://app.cachix.org/billing

so it seems like they raised it.

While Cachix makes it very easy to get up and running now, [setting up our own Nix binary cache](https://nixos.wiki/wiki/Binary_Cache) is easy to do if we so choose. Be it on our own servers or using AWS S3, as we are already doing for the non-Nix builds.